node/service/src/lib.rs

```diff
@@ -122,6 +122,27 @@ impl sc_executor::NativeExecutionDispatch for MoonbaseExecutor {
 	}
 }
 
+/// Trivial enum representing runtime variant
+#[derive(Clone)]
+pub enum RuntimeVariant {
+	Moonbeam,
+	Moonriver,
+	Moonbase,
+	Unrecognized,
+}
+
+impl RuntimeVariant {
+	pub fn from_chain_spec(chain_spec: &Box<dyn ChainSpec>) -> Self {
+		match chain_spec {
+			spec if spec.is_moonbeam() => Self::Moonbeam,
+			spec if spec.is_moonriver() => Self::Moonriver,
+			spec if spec.is_moonbase() => Self::Moonbase,
+			_ => Self::Unrecognized,
+		}
+	}
+}
```

Comment:
This mostly exists because just about everything else related to the runtime variants and chain specs are not `Copy`, `Clone`, etc. I'm open to other suggestions (esp. @librelois)